src/uint256.cpp

```diff
@@ -22,7 +22,7 @@ std::string base_blob<BITS>::GetHex() const
 {
     char psz[sizeof(data) * 2 + 1];
     for (unsigned int i = 0; i < sizeof(data); i++)
-        sprintf(psz + i * 2, "%02x", data[sizeof(data) - i - 1]);
+        snprintf(psz + i * 2, 3, "%02x", data[sizeof(data) - i - 1]);
```

Comment:
> It's correct, and it's also why snprintf isn't an especially safe function call. (replaces one class of issue with another more subtle one)
Indeed, that's why we never use it. The go-to function we have for this is "strprintf". However, it's kind of high overhead for formatting two hex characters. Especially as this function is called a lot.

I'll just replace this with use of `HexDigit` instead.